oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java

```diff
@@ -0,0 +1,406 @@
+		  logMessage(0, "Updating ERROR: " + ex.getMessage());
+		  for (StackTraceElement element : ex.getStackTrace()) {
+			logMessage(0, element.toString());
+		  }
+		}
+	  }
+	  recurseContent(contentLoader, courseToc, getChildren(contentLoader, content.getId()), placementExternalId, level + 1);
+	}
+  }
+
+  private void mapToString(Map<String, String> map, int level, String label) {
+	try {
+	  Iterator<String> i = map.keySet().iterator();
+	  while (i.hasNext()) {
+		String attribute = i.next();
+		logMessage(level, attribute + " : " + map.get(attribute));
+	  }
+	} catch (Exception exc) {
+	  logMessage(level, label + " error : " + exc.getMessage());
+	}
+  }
+
+  protected BbList getChildren(ContentDbLoader contentLoader, Id bbContentId) throws PersistenceException {
+	return contentLoader.loadChildren(bbContentId);
+  }
+
+  private void fixContent(CourseToc courseToc, Content ltiContent, int level, String placementExternalId) throws Exception {
+	equellaLookedAt++;
+
+	ExtendedData extendedData = ltiContent.getExtendedData();
+	//Update the URL
+	String newUrl = getEquellaUrl() + extendedData.getValue("url");
+	logMessage(level, "NewUrl: " + newUrl);
+	ltiContent.setUrl(newUrl);
+	//Update the Host
+	String newUrlHost = getDomainName(getEquellaUrl());
+	logMessage(level, "NewUrlHost: " + newUrlHost);
+	ltiContent.getBbAttributes().setString("UrlHost", newUrlHost);
+	//Update the Type
+	ltiContent.setContentHandler("resource/x-bb-blti-link");
+	//Update the content (remove the link)
+	String htmlContent = ltiContent.getBody().getFormattedText();
+	String toRemove = StringUtils.substringBetween(htmlContent, "<div class=\"equella-link", "</div>");
+	htmlContent = StringUtils.remove(htmlContent,toRemove);
+	FormattedText newFormattedText = new FormattedText(htmlContent,FormattedText.Type.HTML);
+	ltiContent.setBody(newFormattedText);
+	//Update the extended Date
+	Map<String, String> values = new HashMap<>();
+	values.put("customParameters", "");
+	values.put("cimPlacementId", placementExternalId);
+	logMessage(level, "PlacementId: " + placementExternalId);
+	values.put("itemOrigin", "CIM");
+	extendedData.setValues(values);
+	mapToString(values,level, "New Extended Data");
+	// persist it
+	contentDbPersister.persist(ltiContent);
+	fixedItems++;
+
+  }
+
+  private String contentDisplay(Content content) {
+	return "'" + content.getTitle() + "' (" + content.getId().toExternalString() + ")";
+  }
+
+  /**
+   * This is a rather hacky way to get the version.
+   *
+   * @return
+   */
+  @Override
+  protected int getBlackboardVersion() {
+	if (blackboardVersion == 0) {
+	  String vers = BbServiceManager.getConfigurationService().getBbProperty(BbConfig.LIBRARY_VERSION);
+	  // don't use vers.contains (not in Java 1.4)
+	  if (vers.indexOf('.') > -1) //$NON-NLS-1$
+	  {
+		vers = vers.split("\\.")[0]; //$NON-NLS-1$
+	  }
+	  try {
+		blackboardVersion = Integer.parseInt(vers);
+	  } catch (NumberFormatException nfe) {
+		blackboardVersion = 6;
```

Comment:
We should log something if this happens.

Reply:
Not anymore in the code